app/models/checklist_manager.rb

```diff
@@ -0,0 +1,56 @@
+#--------------------------
+#
+# @class ChecklistManager
+#
+# @desc Responsibility: Creates UserChecklists on demand. Reports whether a user has completed a checklist.
+# TODO I'm not sure where these responsibilities should go so I'm putting them here for now. As I develop and refine them,
+# hopefully it'll be clear where the responsibilities should be
+#
+#
+# @author Ashley Engelund (ashley.engelund@gmail.com  weedySeaDragon @ github)
+# @date   12/21/19
+#
+#--------------------------
+
+
+class ChecklistManager
+
+  # Given a checklist (e.g. OrderedListEntry), create a UserChecklist
+  # for the user.  Copy all items in the original list so there is a
+  # UserChecklist entry for every item, including all nested ones.
```

Comment:
" ... all nested ones." - so, this implies that a checklist can contain other checklists(?)

Reply:
Yup.  That's what the `ancestry` gem provides:  nesting for ActiveRecords

Reply:
My question was not technical, but was referencing the design intent of this effort.  For instance, for business subcategories, we've decided that subs can only go one level deep.  We should probably state the design intent (of that particular item) somewhere, probably the model file.

Reply:
I agree.  I've changed the description of the `OrderedAncestryEntry`.  Hopefully this helps:

```
  # @desc Responsibility: This is the master list that is a template for
  #  UserChecklists that are created.  It is the source (blueprint) for the
  #  UserChecklists.
  #  By including OrderedAncestryEntry, it becomes an ordered and nested list.
```